examples/memstore-plugin/main.go

```diff
@@ -45,11 +47,27 @@ func main() {
 	opts := memory.Options{}
 	opts.InitFromViper(v)
 
-	grpc.Serve(&memoryStore{store: memory.NewStore()})
+	plugin := &memoryStore{
+		store:        memory.NewStore(),
+		archiveStore: memory.NewStore(),
+	}
+	grpc.Serve(&config.PluginServices{
+		Store:        plugin,
+		Capabilities: plugin,
+		ArchiveStore: plugin,
+	})
 }
 
 type memoryStore struct {
-	store *memory.Store
+	store        *memory.Store
+	archiveStore *memory.Store
+}
+
+func (ns *memoryStore) Capabilities() (*extra.Capabilities, error) {
```

Comment:
Instead of making the plugin implement this interface can't we just infer capabilities on whether or not it implements the Archive interfaces.

This could be done automatically in Serve/ServeWithGRPCServer methods in `plugin/storage/grpc/grpc.go`

Reply:
I made this refactor. Now capabilities are calculated from archive implementation: https://github.com/jaegertracing/jaeger/pull/2317/files#diff-7df9e55281a47c4dbf643c4aaa7b6895R168